local-cluster/tests/local_cluster.rs

```diff
@@ -3452,10 +3572,10 @@ fn run_test_load_program_accounts(scan_commitment: CommitmentConfig) {
     t_scan.join().unwrap();
 }
 
-fn wait_for_next_snapshot(
+fn wait_for_next_full_snapshot(
```

Comment:
How about coalescing this functionality with `wait_for_next_full_and_incremental_snapshot()` by

1. Creating top level function `wait_for_next_snapshot(wait_for_snapshot: Vec<SnapshotType>)`. This fetches the latest slot via the client and then calls

one or both of:
2. `wait_for_next_incremental_snapshot(last_slot: Slot)`
3. `wait_for_next_full_snapshot(last_slot: Slot)`

Reply:
Yah, should be doable!

Reply:
So the reason why I have the current implementation is because `wait_for_next_incremental_snapshot()` likely does not do what we want it to. It's how I originally did the code, but ran into this problem:

full snapshot interval: 100
incremental snapshot interval: 10

last full snapshot slot: 500
last incremental slot: 590
last_slot: 595

If waiting for next incremental snapshot, which happens at slot 610, it would have a different base slot (full snapshot slot) of 600 and not 500. So then when the function returned, the previous full snapshot and the returned incremental snapshot would be incompatible.

I think I can solve this with an impl very similar to `wait_for_next_full_and_incremental_snapshot()`, but you'll only have two options: either waiting for the next full snapshot, or waiting for the next incremental snapshot (but this one will always return the associated full snapshot as well).

Reply:
Done!